### PR TITLE
LPD-25672 Create endpoints for getting asset library and site document shortcuts.

### PR DESCRIPTION
**Pull request by SamZiemer**
LPD-25672

### What is this trying to solve?
https://liferay.atlassian.net/browse/LPD-25672

Add additional endpoints to the DocumentShortcut headless API

### How am I fixing it?
Creating two additional endpoints to get all Document Shortcuts in a Site, and all Document Shortcuts in an Asset Library

### How can you verify that it works?
Run the tests in DocumentShortcutResourceTest

### CODE REVIEW

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java [hunk 1]
```diff
@@ -38,6 +49,19 @@ public DocumentShortcut getDocumentShortcut(Long documentShortcutId)
...
 	}
 
+	@Override
+	public Page<DocumentShortcut> getSiteDocumentShortcutsPage(Long siteId)
+		throws Exception {
+
+		List<FileShortcut> fileShortcuts = _dlAppService.getGroupFileShortcuts(
+			siteId);
+
+		return Page.of(
```
Comment (jkappler) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java:59]: @SamZiemer, is there no pagination? What if the user has hundreds/thousands of shortcuts?

Comment (SamZiemer) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java:59]: Thanks @jkappler, you make a good point, we definitely need pagination for instances of many many shortcuts, I've added that in some new commits.

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:relevant

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:test:sf

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java [hunk 2]
```diff
@@ -57,9 +85,26 @@ private DocumentShortcut _toDocumentShortcut(FileShortcut fileShortcut)
 				contextUser));
 	}
 
+	private List<DocumentShortcut> _toDocumentShortcuts(
+			List<DLFileShortcut> dlFileShortcuts)
+		throws Exception {
+
+		List<DocumentShortcut> documentShortcuts = new ArrayList<>();
+
+		for (DLFileShortcut dlFileShortcut : dlFileShortcuts) {
```
Comment (jkappler) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java:94]: @SamZiemer, I think we can use TransformUtil here (Brian will likely change this 😅).

**Comment by jkappler** [comment]
Other than the small comment, PR looks good! I will leave the final decision to @jonathanmccann. If he sees all is well we can move forward

**Review by jonathanmccann — APPROVED** [review]

**Comment by jkappler** [comment]
ci:report:2572974715

**Comment by jkappler** [comment]
ci:forward:force

**Comment by jkappler** [comment]
@SamZiemer, build rest issues 😢

**Comment by SamZiemer** [comment]
ci:test:relevant

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java (hunk 2 above)
Comment (SamZiemer) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentShortcutResourceImpl.java:94]: Hopefully he will be ok with this. I tried to stick to a similar case like we have [here](https://github.com/SamZiemer/liferay-portal/blob/master/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java#L933-L943) to keep it as alike other liferay code as I could.

**Comment by SamZiemer** [comment]
@jkappler it looks like the build rest issues are resolved in the tests, do you think this is safe to be force forwarded? Looking at the current failures I do not think the failures are related to the changes I made.

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:stop

**Comment by jkappler** [comment]
ci:report:2573243744

**Comment by jkappler** [comment]
@SamZiemer, looks safe to forward. Ran relevant by mistake 😅.

Once the CI stops, please feel free to forward it.

Thanks!

**Comment by SamZiemer** [comment]
ci:test:sf

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by SamZiemer** [comment]
Build rest failures again, looks like master updated in the meantime. I've added a new commit and going to forward again.

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by SamZiemer** [comment]
ci:forward:force

**Comment by SamZiemer** [comment]
ci:forward:force

**Pull request closed without merging**